kafka-client/src/test/java/dev/responsive/kafka/integration/ResponsiveKafkaStreamsIntegrationTest.java

```diff
@@ -0,0 +1,171 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.producer.KafkaProducer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.streams.StreamsBuilder;
+import org.apache.kafka.streams.kstream.KStream;
+import org.apache.kafka.streams.kstream.Named;
+import org.bson.types.Binary;
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.containers.MongoDBContainer;
+
+public class ResponsiveKafkaStreamsIntegrationTest {
+
+  public static final String COUNT_TABLE_NAME = "count";
+  @RegisterExtension
+  static ResponsiveExtension EXTENSION = new ResponsiveExtension(StorageBackend.MONGO_DB);
+
+  private static final String INPUT_TOPIC = "input";
+  private static final String OUTPUT_TOPIC = "output";
+
+  private final Map<String, Object> responsiveProps = new HashMap<>();
+
+  private String name;
+  private MongoDBContainer mongo;
+
+  @BeforeEach
+  public void before(
+      final TestInfo info,
+      final Admin admin,
+      final MongoDBContainer mongo,
+      @ResponsiveConfigParam final Map<String, Object> responsiveProps
+  ) throws InterruptedException, ExecutionException {
+    // add displayName to name to account for parameterized tests
+    name = info.getTestMethod().orElseThrow().getName() + "-" + new Random().nextInt();
+    this.mongo= mongo;
+
+    this.responsiveProps.putAll(responsiveProps);
+
+    final var result = admin.createTopics(
+        List.of(
+            new NewTopic(inputTopic(), Optional.of(1), Optional.empty()),
+            new NewTopic(outputTopic(), Optional.of(1), Optional.empty())
+        )
+    );
+    result.all().get();
+  }
+
+  private String inputTopic() {
+    return name + "." + INPUT_TOPIC;
+  }
+
+  private String outputTopic() {
+    return name + "." + OUTPUT_TOPIC;
+  }
+
+  @Test
+  public void shouldDefaultToResponsiveStoresWhenUsingDsl() throws Exception {
+    // Given:
+    final List<KeyValueTimestamp<String, String>> inputEvents = Arrays.asList(
+        new KeyValueTimestamp<>("key", "a", 0L),
+        new KeyValueTimestamp<>("key", "b", 2_000L),
+        new KeyValueTimestamp<>("key", "c", 3_000L),
+        new KeyValueTimestamp<>("STOP", "ignored", 18_000L)
+    );
+    final CountDownLatch outputLatch = new CountDownLatch(1);
+
+    final StreamsBuilder builder = new StreamsBuilder();
+    final KStream<String, String> input = builder.stream(inputTopic());
+    input
+        .groupByKey()
+        .count(Named.as(COUNT_TABLE_NAME))
```

Comment:
*Note:* `COUNT_TABLE_NAME` doesn't actually get used as the store name! Pretty crazy and unfortunate because I think you actually need to use `Materialized.as` in order to get the name.

Reply:
Ugh that is unfortunate...why would we not use actual processor names for state stores??? I honestly thought you were mistaken and had to check this for myself, it just seems too stupid. And we can't ever "fix" this without it being a backwards incompatible change. That sucks